buildSrc/src/main/java/com/google/firebase/gradle/ReleaseGenerator.kt

```diff
@@ -101,25 +205,60 @@ open class ReleaseGenerator : DefaultTask() {
   ) =
     repo
       .log()
-      .addPath("$directory/")
+      .addPath(directory)
       .addRange(previousReleaseRef, currentReleaseRef)
       .setMaxCount(1)
       .call()
       .iterator()
       .hasNext()
 
-  private fun writeReleaseConfig(configPath: File, libraries: List<String>, releaseName: String) {
-    File(configPath, "release.cfg")
-      .writeText(
-        """
-                    [release]
-                    name = $releaseName
-                    mode = RELEASE
-                    
-                    [modules]
-                    ${libraries.joinToString("\n".padEnd(21, ' '))}
-                """
-          .trimIndent()
-      )
+  private fun getDirChanges(
+    repo: Git,
+    previousReleaseRef: ObjectId,
+    currentReleaseRef: ObjectId,
+    directory: String
+  ) =
+    repo.log().addPath(directory).addRange(previousReleaseRef, currentReleaseRef).call().map {
+      toCommitDiff(repo, it)
+    }
+
+  private fun toCommitDiff(repo: Git, revCommit: RevCommit): CommitDiff {
```

Comment:
This might make more sense to be a constructor or static method in `CommitDiff`

Reply:
+1